src/praisonai-agents/praisonaiagents/agents/agents.py

```
@@ -1115,6 +1115,30 @@ def _apply_task_guardrail(self, task, task_id, task_output):
             return task_output, True  # Signal retry needed
 
+    def _run_task_start_hook(self, task, task_id):
+        """Run the on_task_start hook and propagate global variables to the task.
+
+        Shared by run_task (sync) and arun_task (async) so the lifecycle hooks
+        and variable propagation stay consistent across both paths.
+        """
+        if self.on_task_start:
+            try:
+                self.on_task_start(task, task_id)
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)